compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmInstructionGen.java

```diff
@@ -1210,7 +1211,8 @@ private void generateBitwiseXorIns(BIRNonTerminator.BinaryOp binaryIns) {
 
         this.mv.visitInsn(LXOR);
 
-        if (!TypeTags.isSignedIntegerTypeTag(opType1.tag) && !TypeTags.isSignedIntegerTypeTag(opType2.tag)) {
+        if (!TypeTags.isSignedIntegerTypeTag(opType1.tag) && !TypeTags.isSignedIntegerTypeTag(opType2.tag) &&
+            opType1.tag != TypeTags.UNION && opType2.tag != TypeTags.UNION) {
```

Comment:
This will be false for `type MyType2 int:Unsigned16|int:Unsigned32;` as well. Not sure how this will have an effect. Thoughts @warunalakshitha ?

Reply:
Should we check all member types in the union and if all member types are unsigned this should be true.

Reply:
I don't this that we need to check all the member types as if the operand type is union type the result of the expression is `int`